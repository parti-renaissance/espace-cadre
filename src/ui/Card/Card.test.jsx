import { render } from '@testing-library/react'
import UICard from './Card'

vi.mock('@mui/system', () => ({
  styled: c => () => c,
}))

vi.mock('@mui/material', async importOriginal => {
  const actual = await importOriginal()
  return {
    ...actual,
    Paper: ({ children }) => <div className="mui-paper-mock">{children}</div>,
    // your mocked methods
  }
})
describe('Card', () => {
  it('displays a Card with headers', () => {
    const { container } = render(<UICard header="header" />)
    expect(container).toMatchSnapshot()
  })

  it('displays a Card with content', () => {
    const { container } = render(<UICard content="content" />)
    expect(container).toMatchSnapshot()
  })

  it('displays a Card with actions', () => {
    const { container } = render(<UICard actions="actions" />)
    expect(container).toMatchSnapshot()
  })
})
