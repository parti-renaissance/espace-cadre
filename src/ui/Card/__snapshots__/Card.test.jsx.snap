// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > displays a Card with actions 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    data-cy="ui-card"
  >
    <div>
      actions
    </div>
  </div>
</div>
`;

exports[`Card > displays a Card with content 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    data-cy="ui-card"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      content
    </div>
  </div>
</div>
`;

exports[`Card > displays a Card with headers 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    data-cy="ui-card"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      header
    </div>
  </div>
</div>
`;
